### DISCUSSION TITLE
Initiatives for the Development Team

### DISCUSSION THREAD

**Discussion by HasithDeAlwis** [cuhacking/2025#45] [Ideas]
Hi everyone! I'd like to start a discussion here about initiatives for cuHacking dev team, we have a lot of talent on our team and should dream big. Let's get down a list of goals and features we want to see for cuHacking 2025 in this discussion thread, I want this to be the hackathon platform other universities gawk about. 

I'll start us off with the following: 
- Project submissions happen on our app, I'm tired of devpost, hackathons are all about open source, and we should have an open-source platform to hand in projects
- Data analytics, let's make sure that we can make informed decisions for cuHacking 2026, 2027 and so on.
- SMS or email notifications for hackers to be aware of their queue in the judging rotation. A hacker's judging happens, makes or breaks a hackathon, I've yet to be in a hackathon where judging has gone smoothly. We have to make sure hackers are aware of their order in the queue and we need an interface for judges to keep track of who they've reviewed. Ideally, hackers can see the queue and know their spot in line. 
- Discord bot authentication. We should have a discord bot that can authenticate users and give them the correct roles when they join the cuHacking community discord. 
- Judging should happen on our platform. No Excel sheets, judges can have a platform to write notes and scores that ALL other judges can see. Potential follow-up idea: Give hackers a chance to see what judges wrote about them, creating a tight feedback loop for participants.

Let's get this show on the road! Leave any ideas here, and we'll have an open discussion about them. Please feel free to criticize or expand upon any of my ideas above, this is only the start.

**Comment by HasithDeAlwis** [comment]
DJ Station! Hacker's can send in song requests and they automatically have it added to a Spotify queue or playlist, we'd need a speaker thats playing in some sort of common area. At the end of the hackathon, we can send our a cuHacking playlist of all the songs submitted by our hackers!

**Comment by MFarabi619** [comment]
We need more pages on our website. Right now it's just a single landing page. The website should hooked up to the CMS so that other teams log in to the dashboard and add/modify content. [Payload CMS](https://payloadcms.com/) is the best on the market right now in my opinion. You can use REST/GRAPHQL/SOAP or whatever you want. Works with both MongoDB and Postgres which is huge. It's also open-source which is huger. We can contribute back and hopefully they'll even feature us on their website.

Highlight Carleton's CS department, SWE, BIT Interactive Multimedia and Design joint degree with Algonquin, Aerospace (Carleton's one of few in Canada that has it). Just like how people want to get into Waterloo because of Hack the North, people should pick Carleton for cuHacking.

Give shoutouts to the STEM clubs like:
Carleton Blueprint
BITSoc
CAIS
Carleton CUSEC Society
SCESoc
Pathways in STEM
IEEE Carleton
Women in Engineering
Carleton Planetery Robotics Team
Blackbird UAV
Astronomy Club
Design League
Math & Stats Society
Physics society
Way more that I'm too lazy to type out and you get the idea

The CUSA Clubs directory page isn't great. The benefit of this is that we're a much better-looking, filtered version of clubs that are relevant to students seeking tech clubs to join. Drives a LOT of traffic to our site, big win during frosh week. Provide these clubs role-based access to our CMS so that they can change their own info without asking us. Provide a toggle to use webscraping to keep their socials and other metadata in sync with the  CUSA Clubs directory, so that they don't have to change it in both places. Write tests for this as well. If there's a mismatch between their profile card on ours vs the CUSA one then they'll get an automated email. This will actually pop off. Also incentivizes their members to collaborate with us

Have an events page that is hooked up to a CMS that other teams can use.

**Comment by MFarabi619** [comment]
Put out regular bug bounties and invite the community to hack into our stuff. Give prizes and social media shoutouts to people that successfully hack into our stuff and embarrass us. Get these people to teach us pentesting and other areas of security that are blind spots to us. Send a joke email to the Cybersecurity club calling them soy devs so that they go ham us. Us vs. them. Could become a friendly rivalry that benefits both parties, builds people's portfolios, secures our platform, and kicks off a tradition; similar to how SCESoc has societal wars. 
Whatever initiatives we're not able to take to completion in this iteration, make them side quests during the cuHacking hackathon that way they're used next year. These people can even join cuHacking to continue working on them.

**Comment by MFarabi619** [comment]
Highlight Ottawa as the Silicon Valley of the North. Feature the Carleton Innovation Hub, KNBA, and Invest Ottawa. Reach out to KNBA to get an SVG version of this image, and render it on our website which each company's logo hyperlinked and clickable. This will drive even more traffic to our site because we can send this to people. The logos being clickable and the image being more easily browsable is a huge UX boost.

![image](https://github.com/user-attachments/assets/83c25440-0781-4c5e-aa19-834ac8604f78)

Consider even like, taking out all the white space around it and just making it a part of the design so that it fits into our website perfectly. Could even be like a "road" going West to the tech park. The user starts on the Carleton campus and as they keep scrolling they end up there. The design already has a similar thing going on with the river.

![image](https://github.com/user-attachments/assets/c45ea86f-56d4-49e2-8efb-318ae9fd4113)

**Comment by MFarabi619** [comment]
Get our design system featured on [Storybook's showcase](https://storybook.js.org/showcase).

**Comment by MFarabi619** [comment]
Make the sponsorship package an interactive web application with a Chatbot/LLM that directs the reader to a section that answers their question. The sponsors will be blown away because we used our technical chops to give them a fun, interactive experience. They've read hundreds of sponsorship packages but one that's a web app? Crazy. It should even be able to allow them to modify their perks, which should auto-recalculate how much their sponsorship would cost and let them know what tier they'd fall into.

Sponsorship package builder 🤯 Just like how you make draft schedules on Carleton Central, they can send us different drafts and proposals. They should have profiles and be able to see a history of all sponsorship packages and metadata from past years of their involvement with cuHacking. Legacy sponsors can even get extra perks and first dibs on stuff before it goes out to other sponsors. There should also be analytics dashboards, government organizations like CSE tend to always request it.

The sponsorship package content should be hooked up to the CMS. The sponsorship team can modify this, push the change, and it'll trigger a pipeline that immediately auto-uploads the most updated version to google drive using the apps script API. The package should be semantically versioned so that confusion and disaster is avoided. The sponsors can also see notifications on their profile that the sponsorship package has changed, and it should take them to a side-by-side diff. They can opt-in to email notifications as well. Super fun project. Can be hosted at sponsorship.cuhacking.ca, and will have role-based auth. Needs extremely thorough security testing. 
Take the DevOps a step further so that when they're confirmed as a sponsor on that app, it'll trigger a pipeline that uploads their image and hyperlink to the corresponding tier on our website using the CMS, and creates a to-do on Notion for marketing. Not a second is wasted.

**Comment by MFarabi619** [comment]
When serving food during the hackathon, have one of the servers be a robot arm. Simple and fun mechatronics project that can be extended with computer vision. Have a little screen next to it that shows the computer vision algorithm running and working. Put a button on it so that people can take pictures. The robot arm photobooth 🤯 

It shows a QR code after that they can scan to download the picture, at which point you're getting into cloud development too. If budget does not allow, the pictures can be printed on A4 paper, or like a cuHacking-branded polaroid; serving as a memory. Give people the option to have the cuHacking logo on it, as some may not want that. 

Provide previews of what it would look like with and without the logo. That way they can re-take the picture if they don't like it; leading to less wasted resources. Also put the option for all these pictures to be featured on a collage in the closing ceremonies slideshow, and to serve as marketing material post-hackathon.

**Comment by MFarabi619** [comment]
During the hackathon, put out recycling challenges. Have the team names on a big, live leaderboard and use webhooks to make kahoot-like announcements on both the discord and hacker portal when a team loses the top spot or something. Have the points allotted by weight of recycled material to keep it simple. All liquids need to be emptied out. Wet paper does not count because it adds weight unnecessarily. You'll see how well and quickly hackathon cleans up after itself 🤣 Boutta have some legendary pictures on Reddit after of people with their bodies halfway inside the recycling bins around campus and sorting them for us lmaooo. Adds to the tradition.

Teams can sneak into other team's rooms to steal their recycling so people need to stick around and protect them, raises the fun even more. Might backfire because the sweats will start going into people's bags to find more garbage and as a result theft rates will go up too so it's a tradeoff.

**Comment by MFarabi619** [comment]
🐧  Have a 'put linux/freebsd on your machine' station where we can provide people with options of what distro/setup they want. We can have a number of machines with different examples so that people can try them interactively before choosing. Have different flash drives with the binaries, and boot into usb. Mentor them through the whole process of installing linux so that they see how easy it is. Get them set up with a nice desktop environment or have a 'ricing' side challenge with judges and prizes. Adds to the open-source initiative. Feature these linux/unix rices on the closing ceremonies slides and use them as marketing material.

Could even be a 'Save your old machine` station where people bring their busted laptops/desktops instead of throwing them out. We don't have to worry about dual booting and can replace the entire OS. Now, they have a whole machine with pure bare-metal linux/freebsd, and we also add to to the hardware sustainability and recycling initiative.

Take this a step further by teaching them how to run it as a web server to host anything they want, including LLMs.

**Comment by MFarabi619** [comment]
Get featured on [Daily.dev](https://daily.dev/), and [Software for Hackathons - MLH](https://guide.mlh.io/organizer-resources/software-for-hackathons). With the quality of our work + documentation, this is 1000% doable.

**Comment by MFarabi619** [comment]
Promote our platform at QA classes so that people contribute to writing test suites. Free and easy contributions for them, they get to learn testing and best practices, and we get to focus more on development and PR reviews. We establish the basic patterns and they just follow them. We can even make them star our project first if they want their PR merged if we want to be a little Apple-y. This incentivizes us to get a high volume of people to contribute to our platform. That external contributors list is going to go ham. A hackathon platform truly built by the very attendees of it.

**Comment by MFarabi619** [comment]
Announce small feature dev and change request stuff like buttons and backgrounds on ig stories. First come first-serve. First one to open a PR gets the review. If a higher quality PR comes in after, that gets merged instead. Incentivizes people to more closely follow our contribution guidelines when opening a PR and leads to less wasted effort with comments and stuff for us. When they get internships their teammates will enjoy working with them more. You can even do AI-generated PR reviews. They can be pre-prompted with our opinions, philosophies, and best practices for things that cannot be automated by linters.

In future iterations of cuHacking, contributions to the platform could be part of the application process.

**Comment by MFarabi619** [comment]
Put out challenges for people to roast our code, website, platform, docs, infra, project, roadmap, design, everything. Funniest roasts get featured and we also get free feedback. Feedback forms on steroids. They should also be able to see roasts from the rest of the community.

**Comment by JowiAoun** [comment]
Have nicely built user profiles that can be shared around with usernames/links/QR codes. We could make a simple version of [Linktree](https://linktr.ee/) for users, where they can build their profile, and add social media links, some description, hackathon projects, whatever.

**Comment by JowiAoun** [comment]
Achievements with descriptions on the user's profile, i.e. "cuHacking 2024", "Recycling Silver Medal", "Leader Spirit" ...

Makes hackers want to pursue achieving these goals to show them off on their profile (read previous comment) for next hackathons, and potentially helps bring back old participants to build on top of previous achievements. Eventually if we decide to have web app notifications (aws will allow easily allow that), hackers can view the achievements they receive, live. I bet there's lots of gamers that will participate, and this sense of "gamifying" work has become very popular recently.
